packages/eslint-plugin/lib/rules/require-media-dimensions.js

```diff
@@ -0,0 +1,70 @@
+const {elementType, hasProp} = require('jsx-ast-utils');
+
+const {docsUrl, polarisComponentFromJSX} = require('../utilities');
+
+const POLARIS_MEDIA_COMPONENTS = ['Image'];
+const STANDARD_MEDIA_ELEMENTS = ['img', 'video'];
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Require explicit dimensions on media elements to prevent layout shift.',
+      category: 'Best Practices',
+      recommended: true,
+      uri: docsUrl('require-media-dimensions'),
+    },
+    schema: {
+      mediaElements: {
+        type: 'array',
+        items: {
+          type: 'string',
+        },
+        uniqueItems: true,
+      },
+    },
+  },
+
+  create(context) {
+    return {
+      JSXElement: (node) => {
+        if (!isKnownMediaElement(node, context)) {
+          return;
+        }
+
+        const elementName = elementType(node.openingElement);
+        const hasWidthProp = hasProp(node.openingElement.attributes, 'width');
```

Comment:
How will this handle scenarios where props are not explicitly passed to the component, such as:

```jsx
const imgProps = {width: 600, height: 400};

return <img {...imgProps} />
```

There may be cases where we want to conditionally change what the values are set to that would make this approach useful, but it doesn't seem like the current linting setup would handle them. If this is a known limitation of the plugin that's okay but we should document it.

Reply:
Good catch! We can actually account for this, I'll work on a patch.

Reply:
Hm, probably best to not go down the rabbit hole of trying to account for all the possible ways the props can be dynamically passed to the component.

I wonder if we can work around an expected convention though like:

**"the spread identifier must be a simple variable defined within the scope of the component for the lint rule to recognize its presence".** 

This'll make it easier to lookup and provide a site for these conditional props to be stored, otherwise the AST can get really complex when one starts to consider expressions like `...someObject.someOtherObject.imgProps`, `...getImgProps()`, `...getObject().someObject.imgProps` etc.

What are your thoughts on having such constraint around unconventional prop assignment vs not supporting it at all?